dartagnan/src/main/java/com/dat3m/dartagnan/utils/printer/Printer.java

```diff
@@ -34,17 +40,29 @@ public String print(Program program){
         }
         result.append(name).append("\n");
 
-        for(Thread thread : program.getThreads()) {
-            if(shouldPrintThread(thread)){
-                appendFunction(thread);
+        if (mode == Mode.THREADS || mode == Mode.ALL) {
+            for (Thread thread : program.getThreads()) {
+                if (shouldPrintThread(thread)) {
+                    appendFunction(thread);
+                }
+            }
+
+            if (!showInitThreads && !program.getThreads().isEmpty()) {
+                result.append("\nSkipping init threads...");
+                result.append("\n...");
+                result.append("\n...");
+                result.append("\n...");
+                result.append("\n");
             }
         }
 
-        for (Function function : program.getFunctions()) {
-            if (function instanceof Thread) {
-                continue;
+        if (mode == Mode.FUNCTIONS || mode == Mode.ALL) {
+            for (Function function : program.getFunctions()) {
+                if (function instanceof Thread) {
+                    continue;
+                }
+                appendFunction(function);
```

Comment:
Can't you simply iterate over all function and if mode is not function, skip those that are not threads?

Reply:
Actually, `program.getFunctions` does not return `Threads`. I wasn't yet sure if I want threads to be in that list or not, so I put a check into `Printer` so it works correctly in either case.
But even if regular functions and threads were in that list, it would be hard to print a replacement for the skipped init threads